With the -nostdinc that we get --with-headers, we need to pull in GCC
4.3's include-fixed subdir too, because that's what contains limits.h
and syslimits.h now.


for  ChangeLog
2008-01-30  Alexandre Oliva  <aoliva@redhat.com>

	* configure.in (SYSINCLUDES): Add GCC's include-fixed after include.
	* configure: Rebuilt.
	
Index: configure
===================================================================
--- configure.orig	2008-01-30 15:40:55.000000000 -0200
+++ configure	2008-01-31 03:24:36.000000000 -0200
@@ -5066,7 +5066,12 @@ echo "$as_me: WARNING:
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  if test -d "${ccheaders}-fixed"; then
+    fixedccheaders="-isystem ${ccheaders}-fixed"
+  else
+    fixedccheaders=
+  fi
+  SYSINCLUDES="-nostdinc -isystem $ccheaders $fixedccheaders \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&5` &&
Index: configure.in
===================================================================
--- configure.in.orig	2008-01-30 15:40:55.000000000 -0200
+++ configure.in	2008-01-31 03:24:30.000000000 -0200
@@ -915,7 +915,12 @@ test -n "$aux_missing" && AC_MSG_WARN([
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  if test -d "${ccheaders}-fixed"; then
+    fixedccheaders="-isystem ${ccheaders}-fixed"
+  else
+    fixedccheaders=
+  fi
+  SYSINCLUDES="-nostdinc -isystem $ccheaders $fixedccheaders \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&AS_MESSAGE_LOG_FD` &&

-- 
Alexandre Oliva         http://www.lsd.ic.unicamp.br/~oliva/
FSF Latin America Board Member         http://www.fsfla.org/
Red Hat Compiler Engineer   aoliva@{redhat.com, gcc.gnu.org}
Free Software Evangelist  oliva@{lsd.ic.unicamp.br, gnu.org}

